Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CoordinatedGraphicsScene.cpp

```diff
@@ -51,9 +51,12 @@ void CoordinatedGraphicsScene::paintToCurrentGLContext(const TransformationMatri
 
     bool sceneHasRunningAnimations = currentRootLayer.applyAnimationsRecursively(MonotonicTime::now());
 
-    bool didChangeClipRect = false;
-    FloatRoundedRect actualClipRect(clipRect);
+    m_textureMapper->beginPainting(flipY ? TextureMapper::FlipY::Yes : TextureMapper::FlipY::No);
+    m_textureMapper->beginClip(TransformationMatrix(), FloatRoundedRect(clipRect));
+
+    std::optional<FloatRoundedRect> rectContainingRegionThatActuallyChanged;
 #if ENABLE(DAMAGE_TRACKING)
+    currentRootLayer.prepareForPainting(*m_textureMapper);
```

Comment:
Don't we need to move this inside the if?

Reply:
assuming the split of `paint()` into `prepareForPainting()` and `paint()` is done only for `ENABLE(DAMAGE_TRACKING)` - it must stay within guards.